src/main/java/graphql/annotations/dataFetchers/MethodDataFetcher.java

```diff
@@ -135,9 +136,51 @@ private Object buildArg(Type p, GraphQLType graphQLType, Object arg) {
         }
     }
 
-    private Object getFieldValue(Object source, String fieldName) throws IllegalAccessException, NoSuchFieldException {
-        Field field = source.getClass().getDeclaredField(fieldName);
-        field.setAccessible(true);
-        return field.get(source);
+    private Object getGraphQLFieldValue(Object source, String fieldName) throws IllegalAccessException, NoSuchFieldException, InvocationTargetException {
```

Comment:
Separate to few methods, one that deals with fields and one that deals with methods